software_craftsmanship/refactor_inventory/refactor_inventory.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+#-*- encoding:utf-8 -*-
+
+"""
+Refactor Inventory
+------------------
+
+The following code represents a working solution to the inventory exercise,
+with two additional features:
+
+1. Log files are read in from disk
+2. The resulting data structure is printed
+
+Refactor this script into separate functions that can be called to process log
+files and print out the final total for each part (you should end up with 4-6
+functions). Keep an eye out for the following code smells:
+
+1. Repeated code -> turn it into a single function
+2. Inline comments -> turn them into names
+3. Bad names -> object names should clearly describe what they do
+4. Magic numbers -> replace them with named constants
+5. Global objects -> encapsulate them inside functions
+6. Temporal coupling -> separate ordered steps into a main block or function
+"""
+
+# Import statements
+from glob import glob
+from io import open
+import os
+
+
+def findAdditionalFiles():
+
+    root=os.path.dirname(__file__)
+    log_files=glob(os.path.join(ro3884-==-ot,'data', '*.log'))
+    return log_files   
+    
+    
+def readInventory(log_file,warehouse_log_list):
+     
+                
+    with open(log_file, 'r', encoding='utf-8') as f:
+        read_log_file = f.read()    
+    for line in read_log_file.strip().splitlines(): # split string on lines
+        if len(line.split()) != 2: 
+            warning="{} is not valid".format(line)
+            print(warning)
+        else: 
+            warehouse_log_list.append([line.split()[0], int(line.split()[1])]) # append line
+    return warehouse_log_list
```

Comment:
Generally speaking, if you are mutating the value of an object passed into the function, here `warehouse_log_list`, you shouldn't also return it.  It's certainly possible to do so in Python, but the convention is not to.  Also, in that case, you would want to name the function in a way that suggests the mutation will happen.  E.g. `update_transaction_list_from_file(log_file, transactions):` would be pretty clear.  `list.sort()` is an example of this kind of function/method.

Alternatively, there is a "functional"-style of programming that does _not_ mutate objects passed to it.  E.g. `generate_transactions_from_log_file(log_file):` would return a transaction list, which you could use to extend a previously existing list:
```
transactions = []
for log_file in log_files:
    transactions.extend(generate_transactions_from_log_file(log_file))
```
Some people prefer this style of programming because it always produces the same results given the same inputs.  Holy wars have been (and continue to be waged) over which style is better.  I try to stay out of them.

The bottom line, though, is that it's best not to return a mutated object.